database/pgsql/layer.go

```diff
@@ -37,10 +37,11 @@ const (
 		SELECT id FROM layer WHERE hash = $1`
 
 	findLayerFeatures = `
-		SELECT f.name, f.version, f.version_format, t.name, lf.detector_id
-			FROM layer_feature AS lf, feature AS f, feature_type AS t
+		SELECT f.name, f.version, f.version_format, t.name, lf.detector_id, ns.name, ns.version_format
+			FROM layer_feature AS lf, feature AS f, feature_type AS t, namespace AS ns
 			WHERE lf.feature_id = f.id
 				AND t.id = f.type
+				AND lf.namespace_id = ns.id
```

Comment:
what if ls.namespace_id is null? there won't be a matched ns.id.

Reply:
Resolved by using Left join